sql/src/main/java/io/crate/metadata/RoutineInfo.java

```diff
@@ -1,36 +1,66 @@
 /*
- * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
  * license agreements.  See the NOTICE file distributed with this work for
- * additional information regarding copyright ownership.  Crate licenses
+ * additional information regarding copyright ownership.  Crate.io licenses
  * this file to you under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.  You may
  * obtain a copy of the License at
  *
- *  http://www.apache.org/licenses/LICENSE-2.0
+ *   http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
  * License for the specific language governing permissions and limitations
  * under the License.
  *
- * However, if you have executed another commercial license agreement
- * with Crate these terms will supersede the license and you may use the
- * software solely pursuant to the terms of the relevant commercial agreement.
+ * To enable or use any of the enterprise features, Crate.io must have given
+ * you permission to enable and use the Enterprise Edition of CrateDB and you
+ * must have a valid Enterprise or Subscription Agreement with Crate.io.  If
+ * you enable or use features that are part of the Enterprise Edition, you
+ * represent and warrant that you have a valid Enterprise or Subscription
+ * Agreement with Crate.io.  Your use of features of the Enterprise Edition
+ * is governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
  */
+
 package io.crate.metadata;
 
 import javax.annotation.Nonnull;
 
 public class RoutineInfo {
-    private String name;
-    private String type;
+    private final String name;
+    private final String type;
+    private final String body;
+    private final String dataType;
+    private final boolean isDeterministic;
+    private final String definition;
+
+    public RoutineInfo(String name,
+                       String type,
+                       String definition,
+                       String body,
```

Comment:
Maybe use `Nonnull` or `Nullable` annotations to clarify that some are optional and some are not